pkg/services/live/live.go

```diff
@@ -245,39 +253,50 @@ func (g *GrafanaLive) GetChannelHandler(channel string) (models.ChannelHandler,
 		return c, nil
 	}
 
-	c, err = g.initChannel(id)
+	c, err := g.initChannel(id)
 	if err != nil {
 		return nil, err
 	}
 	g.channels[channel] = c
 	return c, nil
 }
 
-func (g *GrafanaLive) initChannel(id ChannelIdentifier) (models.ChannelHandler, error) {
-	if id.Scope == "grafana" {
-		p, ok := g.GrafanaScope.Features[id.Namespace]
+// GetChannelNamespace gives threadsafe access to the channel
+func (g *GrafanaLive) GetChannelNamespace(scope string, name string) (models.ChannelNamespaceHandler, error) {
+	if scope == "grafana" {
+		p, ok := g.GrafanaScope.Features[name]
 		if ok {
-			return p.GetHandlerForPath(id.Path)
+			return p, nil
 		}
-		return nil, fmt.Errorf("Unknown feature: %s", id.Namespace)
+		return nil, fmt.Errorf("Unknown feature: %s", name)
```

Comment:
I think quoting helps debugging, in case of spaces in name.
```suggestion
		return nil, fmt.Errorf("Unknown feature: %q", name)
```